src/player.cpp

```diff
@@ -304,6 +305,72 @@ PlayerErrors Player::loadMem(
     return (PlayerErrors)result;
 }
 
+PlayerErrors Player::setBufferStream(
+    unsigned int &hash,
+    unsigned long maxBufferSize,
+    SoLoud::time bufferingTimeNeeds,
+    PCMformat pcmFormat,
+    dartOnBufferingCallback_t onBufferingCallback)
+{
+    if (!mInited)
+        return backendNotInited;
+
+    std::random_device rd;
+    std::mt19937 g(rd());
+    std::uniform_int_distribution<unsigned int> dist(0, INT32_MAX);
+
+    hash = dist(g);
+
+    std::unique_ptr<ActiveSound> newSound = std::make_unique<ActiveSound>();
+    newSound.get()->completeFileName = "";
+    newSound.get()->soundHash = hash;
+
+    newSound.get()->sound = std::make_unique<SoLoud::BufferStream>();
+    newSound.get()->soundType = TYPE_BUFFER_STREAM;
+    static_cast<SoLoud::BufferStream *>(newSound.get()->sound.get())->setBufferStream(
+        this, newSound.get(), maxBufferSize, bufferingTimeNeeds, pcmFormat, onBufferingCallback);
+
+    newSound.get()->filters = std::make_unique<Filters>(&soloud, newSound.get());
+    sounds.push_back(std::move(newSound));
+
+    return noError;
+}
+
+PlayerErrors Player::addAudioDataStream(
+    unsigned int hash,
+    const unsigned char *data,
+    unsigned int aDataLen)
+{
+    auto const s = findByHash(hash);
+
+    if (s == nullptr || s->soundType != TYPE_BUFFER_STREAM)
+        return soundHashNotFound;
+
+    return static_cast<SoLoud::BufferStream *>(s->sound.get())->addData(data, aDataLen);
```

Comment:
[nit] Maybe check for this and raise an error or something if the sound isn't a buffer stream?

Reply:
Done.